### PR TITLE
fix: hide "disconnected" for unknown device connection status

### LINKED ISSUE #493: convert `isConnected` prop in `DeviceNameWithIcon` to `isDisconnected`
DeviceNameWithIcon has a prop `isConnected`. This is an optional prop, but we are [showing the disconnected icon and message on a falsy check of isConnected](https://github.com/digidem/comapeo-mobile/blob/54be6258d27ea8eea162e8c6268039c3a179a8da/src/frontend/sharedComponents/DeviceNameWithIcon.tsx#L62). This is causing the disconnected icon and message to show up when isConnected is undefined and the user is not necessarily disconnected.

## To Do
- [ ] Convert `isConnected` prop to `isDisconnected`
- [ ] convert [isConnected falsy check](https://github.com/digidem/comapeo-mobile/blob/54be6258d27ea8eea162e8c6268039c3a179a8da/src/frontend/sharedComponents/DeviceNameWithIcon.tsx#L62) to isDisconnected truth check
- [ ] convert all instances of `DeviceNameWithIcon` to do isDisconnected check instead.

### PR DESCRIPTION
**Pull request by EvanHahn**
The `<DeviceNameWithIcon>` component takes an optional `isConnected` prop. If it's missing, we treat the device as disconnected, which is not necessarily right.

This change inverts that, hiding the "Disconnected" text in these cases.

Fixes #493.

### CODE REVIEW

**Review by achou11 — APPROVED** [review]
fixes the issue but not particularly thrilled with the solution (mostly around default values for optional props). how about something like a `connectionState?: 'connected' | 'disconnected'` so that the check is more explicit i.e. `connectionState === 'disconnected' && ...`?

**Comment by EvanHahn** [comment]
Done in 86acc928c1306609a87f875b8fd8d6323e334fa3.

File: src/frontend/sharedComponents/DeviceCard.tsx [hunk 1]
```diff
@@ -24,16 +29,29 @@ export const DeviceCard = ({
...
+      isDisconnected = true;
+      break;
+    default:
+      throw new ExhaustivenessError(deviceConnectionStatus);
+  }
+
   return (
     <TouchableOpacity
-      disabled={!onPress || !isConnected}
+      disabled={!onPress || isDisconnected}
```
Comment (achou11) [inline src/frontend/sharedComponents/DeviceCard.tsx:49]: nit: seems sufficient to do `!onPress || deviceConnectionStatus === 'disconnected'` in this case

File: src/frontend/sharedComponents/DeviceNameWithIcon.tsx [hunk 2]
```diff
@@ -59,7 +77,7 @@ export const DeviceNameWithIcon = ({
             {formatMessage(m.thisDevice)}
           </Text>
         )}
-        {!isConnected && (
+        {isDisconnected && (
```
Comment (achou11) [inline src/frontend/sharedComponents/DeviceNameWithIcon.tsx:80]: again, seems sufficient to do `deviceConnectionStatus === 'disconnected'`

**Review by achou11 — APPROVED** [review]
the exhaustiveness check seems unnecessary - what's the rationale for it?

**Comment by EvanHahn** [comment]
The exhaustiveness check is primarily stylistic. I like it because it requires a decision if we change the possible enum values.

For example, imagine we add a new status, `'connecting'`. Without the check, the code will type-check just fine. With the check, it will not, requiring an explicit decision in these parts of the code.

Personally, I prefer to require a decision from future developers, but it's stylistic. Happy to change it.

**Comment by achou11** [comment]
> The exhaustiveness check is primarily stylistic. I like it because it requires a decision if we change the possible enum values.
> 
> For example, imagine we add a new status, `'connecting'`. Without the check, the code will type-check just fine. With the check, it will not, requiring an explicit decision in these parts of the code.
> 
> Personally, I prefer to require a decision from future developers, but it's stylistic. Happy to change it.

ah i see. yeah stylistically not a fan of it but the reasoning makes sense. fine to keep as is then

**Pull request merged**